### PR TITLE
fix warning

### PR DESCRIPTION
**Pull request by Mee-gu**
add integer type for Vec2 constructor and fix warning relates to Vec2.

### CODE REVIEW

File: cocos/2d/CCActionTiledGrid.cpp [hunk 1]
```diff
@@ -623,7 +623,7 @@ void TurnOffTiles::update(float time)
     for (unsigned int i = 0; i < _tilesCount; i++ )
     {
         t = _tilesOrder[i];
-        Vec2 tilePos( (unsigned int)(t / _gridSize.height), t % (unsigned int)_gridSize.height );
+        Vec2 tilePos( (int)(t / _gridSize.height), t % (int)_gridSize.height );
```
Comment (minggo) [inline cocos/2d/CCActionTiledGrid.cpp]: It is better to convert to float.

File: cocos/2d/CCAutoPolygon.cpp [hunk 2]
```diff
@@ -249,13 +249,13 @@ unsigned int AutoPolygon::getSquareValue(unsigned int x, unsigned int y, const R
...
+    Vec2 tl = Vec2((float)x-1, (float)y-1);
     sv += (fixedRect.containsPoint(tl) && getAlphaByPos(tl) > threshold)? 1 : 0;
-    Vec2 tr = Vec2(x, y-1);
+    Vec2 tr = Vec2((float)x, (float)y-1);
     sv += (fixedRect.containsPoint(tr) && getAlphaByPos(tr) > threshold)? 2 : 0;
-    Vec2 bl = Vec2(x-1, y);
+    Vec2 bl = Vec2((float)x-1, (float)y);
     sv += (fixedRect.containsPoint(bl) && getAlphaByPos(bl) > threshold)? 4 : 0;
-    Vec2 br = Vec2(x, y);
+    Vec2 br = Vec2((float)x, (float)y);
```
Comment (minggo) [inline cocos/2d/CCAutoPolygon.cpp]: These converts are not needed, as Vec2 has corresponding constructors.

Comment (minggo) [reply cocos/2d/CCAutoPolygon.cpp]: Using `Vec2 tl(x-1, y-1);` is better.

File: cocos/scripting/lua-bindings/manual/cocos2d/lua_cocos2dx_manual.cpp [hunk 3]
```diff
@@ -2403,7 +2403,7 @@ int tolua_cocos2d_Node_setAnchorPoint(lua_State* tolua_S)
         if (!ok)
             return 0;
 
-        cobj->setAnchorPoint(cocos2d::Vec2(x,y));
+        cobj->setAnchorPoint(cocos2d::Vec2((float)x,(float)y));
```
Comment (minggo) [inline cocos/scripting/lua-bindings/manual/cocos2d/lua_cocos2dx_manual.cpp:2406]: This convert is not needed. The reason is the same as above.

File: cocos/math/Vec2.h [hunk 4]
```diff
@@ -77,6 +77,7 @@ class CC_DLL Vec2
      * @param yy The y coordinate.
      */
     Vec2(float xx, float yy);
+    Vec2(int xx, int yy) : Vec2((float)xx, (float)yy) {}
```
Comment (crazyhappygame) [inline cocos/math/Vec2.h]: Do not like it ... 
What about 
```
Vec2(int xx, float yy)
Vec2(floay xx, int yy)
....
Vec2(AnyOtherNumber xx, AnyOtherNumber yy)
```

I think that better is to fix all places

**Comment by crazyhappygame** [comment]
I think together with removing warnings compiler options should be enabled to catch all possible regression. Below how I did in my CMake

```
if(MSVC)
    target_compile_options(${PROJECT_NAME}
        PRIVATE
       /WX 
       /W0   # /W1 /W2 /W3 /W4 /Wall
       # /permissive-
      # /sdl
)
else()
  target_compile_options(${PROJECT_NAME}
                         PRIVATE 
                                 -Werror
                                 -Wall
                                 # -pedantic
                                 # -pedantic-errors
                                 # -Wextra
                                 # -Wconversion
                                 # -Wsign-conversion
                                 )
```

**Comment by crazyhappygame** [comment]
I really like idea of removing warning but please consider following:
1. Please for each fixed warning enable compiler check. 
http://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html
2. I think that good start is "-Wall -Werror". int/float conversion are tricky please leave it when all others warning will be fixed
3. Please do not use C style casts. use C++ static_cast<float>(intValue)
4. Please do not use casts everywhere consider change types if possible
instead of
```
int value1 = 2;
float value2 = float(value1);
```
maybe following is possible
```
float value1 = 2;
float value2 = value1;
```

**Review by crazyhappygame — CHANGES_REQUESTED** [review]

**Comment by crazyhappygame** [comment]
@minggo @Mee-gu Please discard this PR.
Please start from "-Wall -Werror" compilers warnings

File: cocos/math/Vec2.h (hunk 4 above)
Comment (Mee-gu) [reply cocos/math/Vec2.h]: I think this way is not well designed since the data type may be int, unsigned int, float, double, etc.

**Comment by Mee-gu** [comment]
> @minggo @Mee-gu Please discard this PR.
> Please start from "-Wall -Werror" compilers warnings

@crazyhappygame  we will fix some basic warnings firstly, then make the warnings into error.

File: cocos/math/Vec2.h (hunk 4 above)
Comment (crazyhappygame) [reply cocos/math/Vec2.h]: Below constructor will accept all types with conversion to float
```
template<class T1, class T2>
Vec2(T1 xx, T2 yy)
: Vec2((float)xx, (float)yy) {}
{
}
```
But still I think that that this class should have Vec2(float xx, float yy); only

Comment (Mee-gu) [reply cocos/math/Vec2.h]: yep, we can use the template but the matter is we need always specify the type in angle brackets when using it. I agree that have `Vec2(float xx, float yy)` only is reasonable.

**Comment by Mee-gu** [comment]
> 3\. Please do not use C style casts. use C++ static_cast(intValue)

Is there any difference between C style cast and C++ static_cast<>() for basic types? @crazyhappygame

File: cocos/scripting/lua-bindings/manual/cocos2d/lua_cocos2dx_manual.cpp (hunk 3 above)
Comment (Mee-gu) [reply cocos/scripting/lua-bindings/manual/cocos2d/lua_cocos2dx_manual.cpp:2406]: `x` and `y` are double, need convert to `float`.

File: cocos/math/Vec2.h (hunk 4 above)
Comment (crazyhappygame) [reply cocos/math/Vec2.h]: Thank you for removing int construction

**Comment by crazyhappygame** [comment]
Please read e.g. here about static_cast
https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines-pro-type-cstyle-cast.html

**Comment by crazyhappygame** [comment]
> @crazyhappygame we will fix some basic warnings firstly, then make the warnings  into error.

Usually it is done like that:
1. Enable all warning as error
2. Enable warning 1 by 1

**Comment by minggo** [comment]
@crazyhappygame if we we enable wall warnings as error, then it will block the travis and development. Treat all warnings as error is the final status, that's why we want to fix warnings.

**Comment by crazyhappygame** [comment]
@minggo you can enable "warning as error" and turn on only fixed warnings.

**Comment by minggo** [comment]
Good to know. 
@Mee-gu i think we can do as @crazyhappygame said. Turn on some warnings as error, then fix it.

**Comment by minggo** [comment]
> Usually it is done like that:

> Enable all warning as error
> Enable warning 1 by 1

I think can nod do like this, we can do like:
- 1. turn on some type of warning as error
- 2. fix this error
- 3. loop 1 and 2 until all warnings are fixed

after all, we turn on all warnings as error

**Comment by minggo** [comment]
> Please read e.g. here about static_cast
https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines-pro-type-cstyle-cast.html

@crazyhappygame as i know, c cast and static_cast have the same effect to basic types.

**Pull request merged**